packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,261 @@
+  `,
+
+  small: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 562px;
+  `,
+
+  default: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 700px;
+  `,
+
+  large: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 800px;
+  `,
+
+  xlarge: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 1270px;
+  `,
+};
+
+const closeButton = css`
+  color: ${uiColors.gray.dark1};
+  position: aboslute;
+  float: right;
+  cursor: pointer;
+`;
+
+const escapeKey = 27;
+
+interface ModalProps {
+  /**
+   * Content that will appear inside of the Modal component.
+   */
+  children: React.ReactNode;
+
+  /**
+   * Determines the active state of the modal
+   *
+   * default: `false`
+   */
+  active: boolean;
+
+  /**
+   * Specifies the size of the Modal.
+   *
+   * default: `default`
+   */
+  size?: ModalSize;
+
+  /**
+   * Callback to change the active state of the Modal.
+   *
+   */
+  setActive?: (
+    active: boolean,
+  ) => void | React.Dispatch<SetStateAction<boolean>>;
+
+  /**
+   * Callback to determine whether or not Modal should close when user tries to close it.
+   *
+   */
+  shouldClose?: () => boolean;
+
+  /**
+   * className applied to root overlay div.
+   *
+   */
+  className?: string;
+}
+
+/**
+ * # Modal
+ *
+ *  Modals place content on top of main window.
+ *
+```
+<Modal
+  active
+  size="large"
+  setActive={setActive}
+  shouldClose={() => console.log('Modal is closing now!')}
+  >  
+  Modal content!
+</Modal>
+```
+ * @param props.active Boolean to describe whether or not Modal is active.
+ * @param props.size String to determine size of Modal. ['xxsmall', 'xsmall', 'small', 'default', 'large', 'xlarge']
+ * @param props.setActive Callback to change the active state of Modal.
+ * @param props.children Content to appear inside of Modal container.
+ * @param props.shouldClose Callback to determine whether or not Modal should close when user tries to close it.
+ * @param props.className className applied to overlay div.
+ *
+ */
+function Modal({
+  active = false,
+  size = ModalSize.Default,
+  setActive,
+  children,
+  shouldClose,
+  className,
+  ...rest
+}: ModalProps) {
+  if (!active) {
+    return null;
+  }
+
+  const contentRef = useRef<HTMLDivElement>(null);
+
+  const handleClose = () => {
+    // Don't close modal if shouldClose returns false or no setActive callback is passed
+    if (!setActive || (shouldClose && !shouldClose())) {
+      return;
+    }
+
+    setActive(false);
+  };
+
+  const handleDocumentClick = (e: React.SyntheticEvent) => {
+    if (!contentRef.current) {
+      return;
+    }
+
+    if (!contentRef.current.contains(e.target as Node)) {
+      handleClose();
+    }
+  };
+
+  const handleEscape = useCallback((e: KeyboardEvent) => {
+    if (e.keyCode === escapeKey) {
+      handleClose();
+    }
+  }, []);
+
+  useEventListener('keydown', handleEscape, {
+    options: { once: true },
+  });
+
+  injectGlobal(`
```

Comment:
If this is ran multiple times does it inject this value multiple times?

Also how is this removed when the component unmounts?

Reply:
we actually do not need to be using this anymore so I removed